app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java

```diff
@@ -89,60 +90,57 @@ private boolean operandEnabled(String operand) {
     }
 
     private boolean operationEnabled() {
-        return (operandEnabled(firstOperand) && operandEnabled(secondOperand) && !operator.isEmpty());
+        boolean operationEnabled = true;
+        if (getError() == Error.ERROR_DIVISION_BY_ZERO || getError() == Error.ERROR_INCOMPLETE_OPERATION) {
+            operationEnabled = false;
+        }
+        return operationEnabled;
+    }
+
+
+    private Double makeOperation() {
+        Double result = DEFAULT_RESULT;
+        switch (operator) {
+            case ADD: {
+                result = Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand);
+                break;
+            }
+            case SUB: {
+                result = Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand);
+                break;
+            }
+            case MUL: {
+                result = Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand);
+                break;
+            }
+            case DIV: {
+                result = Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand);
+                break;
+            }
+        }
+        return result;
     }
 
     @Override
     public Double getResult() {
-        Double result = DEFAULT_RESULT;
-        if (!firstOperand.equals(EMPTY_STRING) && !firstOperand.equals(SUB)) {
+        Double result = null;
```

Comment:
why do you use a null here? isn't better to use a 0.0 or something like that?

Reply:
Solved!